tests/test_main.py

```diff
@@ -1,21 +1,75 @@
 # Copyright (c) 2023 Graphcore Ltd. All rights reserved.
 import sys
 import pathlib
-import os
+import pytest
 from . import test_symlink
 from . import testutils
+from graphcore_cloud_tools.paperspace_utils import symlink_datasets_and_caches
 
 
 REPO_ROOT = pathlib.Path(__file__).parents[1].resolve()
 TEST_FILES_DIR = REPO_ROOT / "tests" / "test_files"
 
 
+fake_data = test_symlink.fake_data
 symlink_config = test_symlink.symlink_config
+settings_file = test_symlink.settings_file
+s3_datasets = test_symlink.s3_datasets
+s3_endpoint_url = test_symlink.s3_endpoint_url
 
 
 def test_symlink_command(symlink_config):
+    """Test the support for fuse-overlay symlinks"""
+    testutils.run_command_fail_explicitly(
+        [
+            sys.executable,
+            "-m",
+            "graphcore_cloud_tools",
+            "paperspace",
+            "symlinks",
+            "--config-file",
+            f"{symlink_config}",
+        ],
+        cwd=str(REPO_ROOT),
+    )
+
+
+@pytest.mark.parametrize("legacy", [False, True])
+def test_s3_symlink_command(settings_file, s3_datasets, monkeypatch, legacy):
```

Comment:
Do we need the `test_symlink_command` test above, if this one is also testing the legacy approach?

Reply:
I think so yes, they'll cover similar parts of the code but they're different ways to access the functionality so test slightly different code paths.

It is also a way to document how to use those commands, and proof that those ways of using the commands work.